test/integration/daemonset/daemonset_test.go

```diff
@@ -275,9 +281,7 @@ func validateDaemonSetPodsAndMarkReady(
 ) {
```

Comment:
might be worth it to pass and verify the number of expected terminalPods

Reply:
I don't think so. The controller should treat those pods as if they no longer exist. No point in checking how many there are IMO.

Reply:
Yeah, there is no real need, since we are marking the pods as terminal. It would just make the intent of the assert function clearer.

Reply:
I prefer to keep the diff as small as possible for cherry-picking purposes.

Reply:
ok